dev/com.ibm.ws.microprofile.health.2.0_fat/fat/src/com/ibm/ws/microprofile/health20/fat/DelayAppStartupHealthCheckTest.java

```diff
@@ -194,15 +194,13 @@ public void run() {
                             repeat = false;
                             startServerThread.join();
                         } else if (System.currentTimeMillis() - start_time > time_out) {
-                            List<String> lines = server1.findStringsInFileInLibertyServerRoot("Exiting init function - Thread.sleep completed.", MESSAGE_LOG);
-
+                            List<String> lines = server1.findStringsInFileInLibertyServerRoot("(CWWKZ0001I: Application DelayedHealthCheckApp started)+", MESSAGE_LOG);
                             if (lines.size() == 0) {
-                                log("testReadinessEndpointOnServerStart", "waiting for DelayedServlet sleep to finish.");
-                                server1.waitForStringInLog("Exiting init function - Thread.sleep completed.");
-                                log("testReadinessEndpointOnServerStart", "DelayedServlet sleep finished.");
-                            }
-                            else {
-                                log("testReadinessEndpointOnServerStart", "DelayedServlet sleep finished but timeout still reached.");
+                                log("testReadinessEndpointOnServerStart", "Waiting for Application to start.");
+                                String line = server1.waitForStringInLog("(CWWKZ0001I: Application DelayedHealthCheckApp started)+");
```

Comment:
What if `waitForStringInLog(...)` returns null?  Should you fail the test case when `line == null`?

Reply:
Basically, we are waiting for the application to start up, if the application hasn't started yet within 3 mins, then it would timeout, if the next iteration of the `/health/ready` endpoint hit doesn't return a response code of 200, after the application started message is found. But I agree, if application started message is not available, then it should fail the test case, since then it would be an infinite loop, if for some reason the application never gets started.